cmd/goProbe/config/config.go

```diff
@@ -39,28 +40,28 @@ type validator interface {
 // Config stores goProbe's configuration
 type Config struct {
 	sync.Mutex
-	DB          DBConfig   `json:"db"`
-	Interfaces  Ifaces     `json:"interfaces"`
-	SyslogFlows bool       `json:"syslog_flows"`
-	Logging     LogConfig  `json:"logging"`
-	API         *APIConfig `json:"api"`
+	DB          DBConfig   `json:"db" yaml:"db"`
+	Interfaces  Ifaces     `json:"interfaces" yaml:"interfaces"`
+	SyslogFlows bool       `json:"syslog_flows" yaml:"syslog_flows"`
+	Logging     LogConfig  `json:"logging" yaml:"logging"`
+	API         *APIConfig `json:"api" yaml:"api"`
 }
 
 type DBConfig struct {
-	Path        string      `json:"path"`
-	EncoderType string      `json:"encoder_type"`
-	Permissions fs.FileMode `json:"permissions"`
+	Path        string      `json:"path" yaml:"path"`
+	EncoderType string      `json:"encoder_type" yaml:"encoder_type"`
+	Permissions fs.FileMode `json:"permissions" yaml:"permissions"`
 }
 
 type CaptureConfig struct {
-	Promisc bool `json:"promisc"`
+	Promisc bool `json:"promisc" yaml:"promisc"`
 
 	// used by the ring buffer
 	// RingBufferBlockSize specifies the size of a block, which defines, how many packets
 	// can be held within a block
-	RingBufferBlockSize int `json:"ring_buffer_block_size"`
+	RingBufferBlockSize int `json:"ring_buffer_block_size" yaml:"ring_buffer_block_size"`
```

Comment:
Quick question. My inner organizing tick begs me to make this a struct and call it `RingBuffer` with fields `BlockSize` and `NumBlocks`. Any objections?

Reply:
Sure, go ahead. By now I'm used to fixing a broken config JSON every time I want to run another instance of `goProbe` locally... :rofl: